src/tls13utils.rs

```diff
@@ -536,6 +536,22 @@ pub(crate) fn check_eq_slice(b1: &[U8], b2: &[U8]) -> Result<(), TLSError> {
     }
 }
 
+/// Parse function to check if two slices `b1` and `b2` are of the same
+/// length and agree on all positions, returning a [TLSError] otherwise.
+#[inline(always)]
+pub(crate) fn check_eq_with_slice(b1: &[U8], b2: &[U8], start:usize, end:usize) -> Result<(), TLSError> {
```

Comment:
This would look nicer with a `range` instead of `start` and `end` arguments. Not sure if hax/F* can handle that though.